### PR TITLE
Refactor websocket hooks to follow the same pattern as http hooks, add context pointer

### PR DESCRIPTION
**Pull request by lubeltra**

### CODE REVIEW

File: Source/WebSocket/WinRT/winrt_websocket.cpp [hunk 1]
```diff
@@ -578,7 +581,8 @@ void MessageWebSocketSendMessage(
 
 HRESULT CALLBACK Internal_HCWebSocketDisconnect(
     _In_ HCWebsocketHandle websocket,
-    _In_ HCWebSocketCloseStatus closeStatus
+    _In_ HCWebSocketCloseStatus closeStatus,
+    _In_opt_ void* context
```
Comment (jplafonta) [inline Source/WebSocket/WinRT/winrt_websocket.cpp]: does this change not need to be made to winhttp_websocket.cpp as well?

Comment (lubeltra) [reply Source/WebSocket/WinRT/winrt_websocket.cpp]: I don't think it's actually being build by anyone, so I missed it

Comment (jplafonta) [reply Source/WebSocket/WinRT/winrt_websocket.cpp]: I believe its used by gsdk, and optionally on win32

---
In reply to: [278250082](https://github.com/Microsoft/libHttpClient/pull/399#discussion_r278250082) [](ancestors = 278250082)

File: Source/WebSocket/Websocketpp/websocketpp_websocket.cpp [hunk 2]
```diff
@@ -809,7 +811,9 @@ HRESULT CALLBACK Internal_HCWebSocketSendBinaryMessageAsync(
     _In_ HCWebsocketHandle websocket,
     _In_reads_bytes_(payloadSize) const uint8_t* payloadBytes,
     _In_ uint32_t payloadSize,
-    _Inout_ XAsyncBlock* asyncBlock)
+    _Inout_ XAsyncBlock* asyncBlock,
+    _In_opt_ void* context
```
Comment (jplafonta) [inline Source/WebSocket/Websocketpp/websocketpp_websocket.cpp]: Also shouldn't the implementation be updated so this actually gets passed back to the callback? here and elsewhere?

Comment (lubeltra) [reply Source/WebSocket/Websocketpp/websocketpp_websocket.cpp]: no this context is for this function, it's not what the consumer registers with their own callbacks (that comes out in get event handlers)

**Pull request merged**